tests/devices/test_default_qutrit_mixed.py

```diff
@@ -1270,3 +1270,320 @@ def qfunc(x, y):
         grad_jit = jax.grad(qnode_jit)(x, y)
 
         assert qml.math.allclose(grad, grad_jit)
+
+
+@pytest.mark.parametrize("nr_wires", [2, 3])
+class TestReadoutError:
+    """Tests for measurement readout error"""
+
+    setup_unitary = np.array(
+        [
+            [1 / np.sqrt(2), 1 / np.sqrt(3), 1 / np.sqrt(6)],
+            [np.sqrt(2 / 29), np.sqrt(3 / 29), -2 * np.sqrt(6 / 29)],
+            [-5 / np.sqrt(58), 7 / np.sqrt(87), 1 / np.sqrt(174)],
+        ]
+    ).T
+
+    def setup_state(self, nr_wires):
+        """Sets up a basic state used for testing."""
+        qml.QutritUnitary(self.setup_unitary, wires=0)
+        qml.QutritUnitary(self.setup_unitary, wires=1)
+        if nr_wires == 3:
+            qml.TAdd(wires=(0, 2))
+
+    @staticmethod
+    def get_expected_dm(num_wires):
+        """Gets the expected density matrix of the circuit for the first num_wires"""
+        state = np.array([2, 3, 6], dtype=complex) ** -(1 / 2)
+        if num_wires == 2:
+            state = np.kron(state, state)
+        if num_wires == 3:
+            state = sum(
+                [
+                    state[i] * reduce(np.kron, [v, state, v])
+                    for i, v in enumerate([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+                ]
+            )
+        return np.outer(state, state)
+
+    # Set up the sets of probabilities that are inputted as the readout errors.
+    relax_and_misclass = [
+        [(0, 0, 0), (0, 0, 0)],
+        [None, (1, 0, 0)],
+        [None, (0, 1, 0)],
+        [None, (0, 0, 1)],
+        [(1, 0, 0), None],
+        [(0, 1, 0), None],
+        [(1, 0, 1), None],
+        [(0, 1, 0), (0, 0, 1)],
+        [None, (0.1, 0.2, 0.4)],
+        [(0.2, 0.1, 0.3), None],
+        [(0.2, 0.1, 0.4), (0.1, 0.2, 0.5)],
+    ]
+
+    # Expected probabilities of measuring each state after the above readout errors are applied.
+    expected_probs = [
+        [1 / 2, 1 / 3, 1 / 6],
+        [1 / 3, 1 / 2, 1 / 6],
+        [1 / 6, 1 / 3, 1 / 2],
+        [1 / 2, 1 / 6, 1 / 3],
+        [5 / 6, 0, 1 / 6],
+        [2 / 3, 1 / 3, 0],
+        [5 / 6, 1 / 6, 0],
+        [2 / 3, 0, 1 / 3],
+        [5 / 12, 17 / 60, 0.3],
+        [7 / 12, 19 / 60, 0.1],
+        [11 / 24, 7 / 30, 37 / 120],
+    ]
+
+    @pytest.mark.parametrize(
+        "relax_and_misclass, expected", zip(relax_and_misclass, expected_probs)
+    )
+    def test_readout_probs(self, nr_wires, relax_and_misclass, expected):
```

Comment:
```suggestion
    def test_probs_with_readout_error(self, nr_wires, relax_and_misclass, expected):
```

**nitpick**: ambiguous test name 

This is minor, but because the readout error arguments for the device are also `_probs`, it's not clear whether the  test relates to those input probs, or the output measurement probabilities of the QNode. Suggested test name is intended to distinguish between the two. Similar naming can be used for the other tests for clarity.